src/compiler.ts

```diff
@@ -8800,7 +8811,7 @@ export class Compiler extends DiagnosticEmitter {
         }
         values[i] = expr;
       } else {
-        values[i] = this.makeZero(elementType);
+        values[i] = this.makeZero(elementType, expression);
```

Comment:
Not directly related to this PR, but looks like we are missing a node represent an omitted expression so we can reference a better code location in diagnostics. I guess we should tackle that in a separate PR and merge it into this one so diagnostics aren't overly surprising.

Reply:
Changed my mind. Going to merge as-is and look into omitted nodes in a follow-up.